### PR TITLE
Orca tf estimator add grad clip support

### PR DESCRIPTION
**Pull request by jenniew**
Orca tf estimator add grad clip support for tf graph model.

### CODE REVIEW

File: pyzoo/zoo/orca/learn/tf/estimator.py [hunk 1]
```diff
@@ -170,7 +173,20 @@ def __init__(self, *, inputs, outputs, labels, loss,
...
-            self.train_op = self.optimizer.minimize(self.loss)
+            if clip_norm or clip_value:
+                gvs = self.optimizer.compute_gradients(self.loss)
+                if clip_norm:
+                    gvs = [(tf.clip_by_norm(g_v[0], clip_norm), g_v[1]) for g_v in gvs]
+                if clip_value:
+                    if isinstance(clip_value, tuple):
+                        gvs = [(tf.clip_by_value(g_v[0], clip_value[0], clip_value[1]), g_v[1]) for g_v in gvs]
+                    if isinstance(clip_value, (int, float)):
+                        gvs = [(tf.clip_by_value(g_v[0], -clip_value, clip_value), g_v[1]) for g_v in gvs]
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/tf/estimator.py:184]: need to check `clip_value` is larger than 0

File: pyzoo/zoo/orca/learn/tf/estimator.py [hunk 2]
```diff
@@ -170,7 +173,20 @@ def __init__(self, *, inputs, outputs, labels, loss,
...
                 "it should be an instance of tf.train.Optimizer"
             self.optimizer = ZooOptimizer(optimizer)
-            self.train_op = self.optimizer.minimize(self.loss)
+            if clip_norm or clip_value:
+                gvs = self.optimizer.compute_gradients(self.loss)
+                if clip_norm:
+                    gvs = [(tf.clip_by_norm(g_v[0], clip_norm), g_v[1]) for g_v in gvs]
+                if clip_value:
+                    if isinstance(clip_value, tuple):
+                        gvs = [(tf.clip_by_value(g_v[0], clip_value[0], clip_value[1]), g_v[1]) for g_v in gvs]
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/tf/estimator.py:182]: need to check `clip_value[1]` is larger than `clip_value[0]`

File: pyzoo/zoo/orca/learn/tf/estimator.py (hunk 1 above)
Comment (jenniew) [reply pyzoo/zoo/orca/learn/tf/estimator.py]: updated

File: pyzoo/zoo/orca/learn/tf/estimator.py (hunk 2 above)
Comment (jenniew) [reply pyzoo/zoo/orca/learn/tf/estimator.py]: updated.

**Comment by jenniew** [comment]
Jenkins pass: http://10.239.47.210:18888/view/ZOO-PR/job/ZOO-PR-Validation/3805/

**Pull request merged**